### ISSUE TITLE
Release 2024.10.0

### ISSUE THREAD

**Issue by fjetter** [dask/community#401]
|  |  |
| --- | --- |
| Release version | 2024.10.0 |
| Planned release date | 2024-10-16 |
| Status | Off schedule |
| Release manager | @fjetter |

## Best effort

_Try to close before the release but will not block the release_

- [ ] ...
- [ ] ...

## Blocker

_Issues that would cause us to block and postpone the release if not fixed_

- [ ] ...
- [ ] ...

## Comments

We're a little off schedule here. We didn't release last friday but would like to cut another release *before* we merge a possibly [impactful PR](https://github.com/dask/distributed/pull/8797) in distributed. Therefore, I'm planning to cut a release tomorrow if there are no objections

**Comment by fjetter** [comment]
done

**Comment by pentschev** [comment]
I just noticed the new release is still not in conda-forge, even though the @fjetter merged the release https://github.com/conda-forge/dask-core-feedstock/pull/219 . I then went to investigate a bit and found that Distributed didn't get a similar PR, which was previously done manually by @jrbourbeau (see https://github.com/conda-forge/distributed-feedstock/pull/293 for example) instead of automatically by the conda-forge bot. I don't know if that's intentional or not that this feature is disabled in the https://github.com/conda-forge/distributed-feedstock , so it's one thing to keep in mind for future releases. In the meantime I opened https://github.com/conda-forge/distributed-feedstock/pull/294 to take care of the 2024.10.0 release, but I need someone to approve/merge, since I'm not authorized to do it there. @fjetter @charlesbluca  would one of you mind having a look?

**Comment by charlesbluca** [comment]
Thanks @pentschev! Left some comments, happy to approve/merge when things are passing. Should also note that once these packages are in and published we should open a similar PR in [dask-feedstock](https://github.com/conda-forge/dask-feedstock) to cut a new release of the `dask` metapackage

> I don't know if that's intentional or not that this feature is disabled

Haven't followed the feedstocks closely, but if I could guess I'd imagine this might've been turned off in `distributed` and `dask`'s feedstocks since they're dependent on `dask-core` and `distributed`, respectively, and the auto-generated PRs would fail at time of opening since they need to be merged and built in sequential order (dask-core, distributed, dask)

**Comment by pentschev** [comment]
> Should also note that once these packages are in and published we should open a similar PR in [dask-feedstock](https://github.com/conda-forge/dask-feedstock) to cut a new release of the `dask` metapackage

Ah! Missed that, sorry, but I see you're doing it in https://github.com/conda-forge/dask-feedstock/pull/274, thanks for addressing that.

**Comment by charlesbluca** [comment]
>  built in sequential order (dask-core, distributed, dask)

I was incorrect here, I'm forgetting `dask-expr` which is a dependency of the metapackage and hard pinned to a specific `dask-core` version, so the order is actually:

- dask-core
- distributed and dask-expr in parallel
- dask [metapackage]

**Comment by pentschev** [comment]
Nice catch, I didn't realize there has been a `dask-expr` release as well since it doesn't follow the same calver and wasn't mentioned here.

**Comment by charlesbluca** [comment]
Opened a PR for dask-expr: https://github.com/conda-forge/dask-expr-feedstock/pull/79

I'm not a maintainer there so will need to wait until that gets approved for us to complete the conda-forge release
